prime-router/src/main/kotlin/history/azure/DeliveryFacade.kt

```diff
@@ -53,6 +57,7 @@ class DeliveryFacade(
         require(since == null || until == null || until > since) {
             "End date must be after start date."
         }
+        val reportId = if (reportIdStr != null) UUID.fromString(reportIdStr) else null
```

Comment:
You might want to put this in a try/catch since `UUID.fromString` will throw if it's not a valid UUID